defs/jit.mk

```diff
@@ -40,6 +40,7 @@ else ifneq ($(strip $(RLIB_DIR)),) # combo build
 $(RUST_LIB): $(srcdir)/ruby.rs
 	$(ECHO) 'building $(@F)'
 	$(gnumake_recursive)$(Q) $(RUSTC) --edition=2024 \
+	    $(JIT_RUST_OPTFLAGS) \
```

Comment:
I think we should rename this `RUSTFLAGS` and have the corresponding lowercase version. It's already not just optimization flags (`-g`) and the ruby crate might gain non-JIT rust code in the future.

Reply:
I tried general RUSTFLAGS and it gets taken over by `cargo` at some point affecting mmtk CI due to some dependency having `-C embed-bitcode=no` which is incompatible with any LTO https://github.com/jackorp/ruby/actions/runs/20429672830/job/58697416079#step:14:143 , and JIT_RUST_FLAGS already exist. 

I think it would be nice to have full RUSTFLAGS, but at I cannot do necessary work in time for Ruby 4.0.